### ISSUE TITLE
ig: Events from the host should be shown by default

### ISSUE THREAD

**Issue by mauriciovasquezbernal** [inspektor-gadget/inspektor-gadget#3419]
#1741 introduced the `--host` flag to be able to show also events coming from the host. The current behavior is:

```bash 
# show only events from containers
$ ig run foo 

# show all events (including containers)
$ ig run foo --host

# show events from specific containers
$ ig run foo --containername mycontainer
```

We discussed the UX of it on https://github.com/inspektor-gadget/inspektor-gadget/pull/1741#issuecomment-1599176460, but I now feel we got it wrong. 

IMO it should work as:

```bash
# show all events, containers, host, etc. 
$ ig run foo 

# show events from specific containers
$ ig run foo --containername  mycontainer 
```

And then, I'll propose to handle other cases with the -F flag. Implementing an offload mechanism to filter those events in eBPF later on:

```bash
# show events not coming from a container
$ ig run foo -F runtime.ContainerID=="" 

# show events coming only from containers
$ ig run foo -F runtime.ContainerID!="" 
```

The `host` flag would be removed. 

/cc @blixtra

**Comment by flyth** [comment]
I agree with not showing host info by default is confusing. But maybe instead of

```sh
$ ig run foo -F runtime.ContainerID=="" 
```

invert the `--host` so that it would become

```sh
$ ig run foo --host-only
```

`-F runtime.ContainerID==""` somehow feels prone to error - what if enrichment fails (somehow) and container ids are empty, but events are still coming from containers? Even if the filter expression would be converted to something that can be offloaded for a host/no-host decision to ebpf (we're just checking for mntnsid == 1, right?), this might be confusing.

**Comment by blanquicet** [comment]
> $ ig run foo -F runtime.ContainerID=="" 
> $ ig run foo -F runtime.ContainerID!="" 
> 
For me this is not very user friendly. I'd prefer to have dedicated flags for this.

**Comment by mauriciovasquezbernal** [comment]
How often will somebody need to filter by events coming only from containers or only from the host? Is it that common? 

Let's ping @blixtra on this one, because one idea was to unify all the filtering having only --filter, i.e. possibly removing --containername, --podname, etc.

**Comment by dorser** [comment]
As a user, events from all containers but not from the host is something we need. As someone who is also building gadgets it aligns with `gadget_should_discard_mntns_id()`

**Comment by mauriciovasquezbernal** [comment]
Perhaps a `--container-only` flag will work:

```bash 
# prints all events
$ ig run foo 

# prints events from all containers (current behavior when usingig run foo)
$ ig run foo --containers-only 

# we don't provide any way to print "host only" events.
```

Any opinions on this?

**Comment by alban** [comment]
Do you mean this?

| Command | Include<br>mount namespaces from<br>host (/host/proc/1/ns/mnt) | Include<br>mount namespaces from<br>recognised containers | Include<br>mount namespaces from<br>unknown sources:<br>1. systemd services<br>   with separate mountns<br>2. host under the<br>   minikube container<br>3. bubblewrap, systemd-nspawn<br>4. etc. |
|---|---|---|---|
| ig run foo | ✅ | ✅ | ✅ |
| ig run foo --containers-only | ❌ | ✅ | ❌ |

**Comment by mauriciovasquezbernal** [comment]
> Do you mean this?

Exactly.

We can say that "ig run foo" shows all events (disabling filtering by mount namespace completely)
